superdesk/publish/publish_queue.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import logging
+
+from superdesk import get_resource_service
+from superdesk.notification import push_notification
+from superdesk.resource import Resource
+from superdesk.services import BaseService
+
+logger = logging.getLogger(__name__)
+
+
+class PublishQueueResource(Resource):
+    schema = {
+        'item_id': Resource.rel('archive', type='string'),
+        'item_version': {'type': 'integer', 'nullable': False},
+
+        'formatted_item': {'type': 'string', 'nullable': False},
+        'subscriber_id': Resource.rel('subscribers'),
+        'destination': {
+            'type': 'dict',
+            'schema': {
+                'name': {'type': 'string', 'required': True, 'empty': False},
+                'format': {'type': 'string', 'required': True},
+                'delivery_type': {'type': 'string', 'required': True},
+                'config': {'type': 'dict'}
+            }
+        },
+        'published_seq_num': {
+            'type': 'integer'
+        },
+        'publish_schedule': {
+            'type': 'datetime'
+        },
+        'publishing_action': {
+            'type': 'string'
+        },
+        'unique_name': {
+            'type': 'string'
+        },
+        'content_type': {
+            'type': 'string'
+        },
+        'headline': {
+            'type': 'string'
+        },
+
+        'transmit_started_at': {
+            'type': 'datetime'
+        },
+        'completed_at': {
+            'type': 'datetime'
+        },
+        'state': {
+            'type': 'string',
+            'allowed': ['pending', 'in-progress', 'success', 'canceled', 'error'],
+            'nullable': False
+        },
+        'error_message': {
+            'type': 'string'
+        }
+    }
+
+    additional_lookup = {
+        'url': 'regex("[\w,.:-]+")',
+        'field': 'item_id'
+    }
+
+    datasource = {'default_sort': [('_created', -1)]}
+    privileges = {'POST': 'publish_queue', 'PATCH': 'publish_queue'}
+
+
+class PublishQueueService(BaseService):
```

Comment:
We can, I've just moved the code from superdesk repo without doing code changes so it's easier to follow, I can do the suggested changes though.